source/concepts/cryptography.rst

```diff
@@ -2,97 +2,78 @@
+* **Voting**: Required for nodes participating in the :ref:`finalization` process.
+* **Transport**: This key pair is used by nodes for secure transport over `TLS <https://en.wikipedia.org/wiki/Transport_Layer_Security>`__.
 
-The following steps are performed to |implementation-public-key-address|:
+.. note::
 
-.. code-block:: cpp
+   As a rule of thumb, the **private key** in any key pair should be kept secret at all times. However, **how bad is it to have a private key stolen?**
 
-	Address PublicKeyToAddress(const Key& publicKey, NetworkIdentifier networkIdentifier) {
-		// step 1: sha3 hash of the public key
-		Hash256 publicKeyHash;
-		crypto::Sha3_256(publicKey, publicKeyHash);
+   .. csv-table::
+      :header: "Key", "Severity", "Impact"
+      :widths: 20,20,60
+      :delim: ;
 
-		// step 2: ripemd160 hash of (1)
-		Address decoded;
-		crypto::Ripemd160(publicKeyHash, reinterpret_cast<Hash160&>(decoded[1]));
+      **Main**; 🔴 HIGH; Funds could be transferred to another account.
+      **Remote**; 🟠 MED; Harmless to the account or the node. Easily reverted by linking another remote account. An attacker grabbing a large number of remote keys    could    gain a lot of harvesting power, influencing which blocks are added to the blockchain.
+      **VRF**; 🟡 LOW; Harmless without the key used for harvesting.
+      **Voting**; 🟡 LOW; Harmless to the account or the node. Easily reverted by linking another voting account. An attacker grabbing more than 50% of the network's       voting keys could influence block finalization.
+      **Transport**; 🟡 LOW; An attacker could steal harvesting delegations away from the node, but harmless otherwise.
 
-		// step 3: add network identifier byte in front of (2)
-		decoded[0] = utils::to_underlying_type(networkIdentifier);
-
-		// step 4: concatenate (3) and the checksum of (3)
-		Hash256 step3Hash;
-		crypto::Sha3_256(RawBuffer{ decoded.data(), Hash160::Size + 1 }, step3Hash);
-		std::copy(step3Hash.cbegin(), step3Hash.cbegin() + Checksum_Size, decoded.begin() + Hash160::Size + 1);
-
-		return decoded;
-	}
-
-As you can see, it is possible to create an address without interacting with the blockchain.
-In fact, the blockchain only tracks addresses and public keys when they first appear in one transaction.
-
-.. |edwards| raw:: html
+*********
+Signature
+*********
 
-   <a href="https://en.wikipedia.org/wiki/Twisted_Edwards_curve" target="_blank">Twisted Edwards curve</a>
+**Messages can be signed** using a private key, producing 512-bit (128 byte) **signatures**.
 
-.. |Ed25519| raw:: html
+Using the matching public key signatures can then **validate** that the key pair signed an entity.
 
-   <a href="https://ed25519.cr.yp.to/" target="_blank">Ed25519</a>
+The implementation can be found in the `Signer class <https://github.com/nemtech/catapult-server/blob/main/src/catapult/crypto/Signer.cpp>`__ under the ``crypto`` module of :doc:`catapult-server <../server>`.
```

Comment:
same comment about donna

Reply:
I somewhat agree. Signer could stay, as we're only using scalarmult and few others, but would reword it, as calling it `implementation` is a) overstatement b) it suggest it might be custom implementation, when it is not.

Reply:
also, if you're mentioning this, it is weird, not to mention batched signature verification.

Reply:
OK, I kept the links to the source because they were in the original version so I thought they must have brought value at some point. But the code structure is complex (with wrappers and all) and this is not the place to explain it so... I think we're better off without the code links. Interested readers will know where to look for the code.